template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/CmTemplateProcessor.java

```diff
@@ -243,7 +247,31 @@ private Map<String, Set<ServiceComponent>> getNonGatewayServicesByHostGroup() {
                                 .filter(i -> !i.getComponent().equalsIgnoreCase("GATEWAY"))
                                 .collect(toSet())
                 ));
+    }
+
+    public List<String> getUnhealthyDependentHostGroups(StackDto stackdto, Set<String> dependentComponent, String undefinedDependency) {
+        Set<String> dependentHostGroupNames = getNonGatewayComponentsByHostGroup().entrySet().stream()
+                .filter(hgComponents ->
+                        isDependentComponentPresentInHostGroup(hgComponents.getValue(), dependentComponent))
```

Comment:
Can this be replaced by dependentComponent.contains(hgComponent) ?

Reply:
hgComponents is set of values. So, we have made that function.